logger/drain/factory_test.go

```diff
@@ -0,0 +1,66 @@
+package drain
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestGetUsingInvalidValues(t *testing.T) {
+	drainStrs := []string{"bogus"}
+	for _, drainStr := range drainStrs {
+		_, err := NewDrain(drainStr)
+		if err == nil ||
```

Comment:
That's Python though :+1: It's not wise to apply style guide of one language on the other.